PolyEngine/Engine/Src/ComponentBase.hpp

```diff
@@ -19,9 +29,13 @@ namespace Poly {
 			return Owner->GetComponent<T>();
 		}
 
+		void SetFlags(const EnumFlags<eComponentBaseFlags>& rhs) { Flags = rhs; }
```

Comment:
This method should call `|=` operator on Flags as this performs bit or and won't clear already set bits.

What's More wee need ResetFlags Method whic would call `Flags &= ~rhs;` to clear the flags provided.